ui/components/app/snaps/snap-install-warning/snap-install-warning.js

```diff
@@ -29,118 +41,117 @@ const checkboxStateReducer = produce((state, action) => {
       state[action.checkboxId] = state[action.checkboxId]
         ? !state[action.checkboxId]
         : true;
-
       break;
     default:
       throw new Error(
         'You must provide a type when dispatching an action for checkboxState',
       );
   }
 });
-
 export default function SnapInstallWarning({
   onCancel,
   onSubmit,
   warnings,
   snapName,
+  isOpen,
 }) {
   const t = useI18nContext();
   const [checkboxState, dispatch] = useReducer(checkboxStateReducer, {});
-
   const isAllChecked = warnings.every((warning) => checkboxState[warning.id]);
-
   const onCheckboxClicked = useCallback((checkboxId) => {
     dispatch({ type: 'check', checkboxId });
   }, []);
 
-  const SnapInstallWarningFooter = () => {
-    return (
-      <div className="snap-install-warning__footer">
+  return (
+    <Modal className="snap-install-warning" onClose={onCancel} isOpen={isOpen}>
+      <ModalOverlay />
+      <ModalContent
+        modalDialogProps={{
+          display: Display.Flex,
+          flexDirection: FlexDirection.Column,
+          gap: 4,
+        }}
+      >
+        <ModalHeader
+          onClose={onCancel}
+          childrenWrapperProps={{
+            display: Display.Flex,
+            flexDirection: FlexDirection.Column,
+            alignItems: AlignItems.center,
+            justifyContent: JustifyContent.Center,
+            gap: 4,
+          }}
+        >
+          <AvatarIcon
+            iconName={IconName.Danger}
+            color={IconColor.warningDefault}
+            backgroundColor={BackgroundColor.warningMuted}
+            size={AvatarIconSize.Xl}
+          />
+          <Text variant={TextVariant.headingMd} textAlign={TextAlign.Center}>
+            {t('snapInstallWarningHeading')}
+          </Text>
+        </ModalHeader>
+        <Text textAlign={TextAlign.Center} marginBottom={4}>
+          {warnings.length > 1
+            ? t('snapInstallWarningCheckPlural', [
+                <Text
+                  key="snapNameInWarningDescription"
+                  fontWeight={FontWeight.Medium}
+                  as="span"
+                >
+                  {snapName}
+                </Text>,
+              ])
+            : t('snapInstallWarningCheck', [
+                <Text
+                  key="snapNameInWarningDescription"
+                  fontWeight={FontWeight.Medium}
+                  as="span"
+                >
+                  {snapName}
+                </Text>,
+              ])}
+        </Text>
+        <Box
+          display={Display.Flex}
+          flexDirection={FlexDirection.Column}
+          gap={4}
+          marginBottom={4}
+        >
+          {warnings.map((warning) => (
+            <Box
+              key={warning.id}
+              display={Display.Flex}
+              gap={2}
```

Comment:
If `gap` represents the padding / margin between the checkbox and the label, we should try increasing it to **4** or potentially **6**

cc @david0xd

Reply:
I think it should match the previous version (at the best possible level).

This is on the latest Flask published:
![Screenshot 2023-08-25 at 10 29 03](https://github.com/MetaMask/metamask-extension/assets/13301024/6c0f2805-0443-49be-aef2-8ff391fc2d6d)

And on this PR:
![Screenshot 2023-08-25 at 10 27 25](https://github.com/MetaMask/metamask-extension/assets/13301024/a6a9ba2b-a7d0-410f-96d0-437012641ccb)

Reply:
Good catch with the checkbox and label spacing and alignment. It should be aligned to the top and the spacing should be 16px(`4`) I've updated it.